test/index.js

```diff
@@ -80,99 +108,115 @@ test('circles and proto option – deep circular array', async ({same, is, isNot
   })
-  test(label + ' – nested async function', async ({is}) => {
+  test(label + ' – nested async function', async ({ is }) => {
     const fn = async () => {}
-    is(clone({fn}).fn, fn, 'same function')
+    is(clone({ fn }).fn, fn, 'same function')
   })
-  test(label + ' – nested generator function', async ({is}) => {
+  test(label + ' – nested generator function', async ({ is }) => {
     const fn = function * () {}
-    is(clone({fn}).fn, fn, 'same function')
+    is(clone({ fn }).fn, fn, 'same function')
   })
-  test(label + ' – nested date', async ({is, isNot}) => {
+  test(label + ' – nested date', async ({ is, isNot }) => {
     const date = new Date()
-    is(+clone({d: date}).d, +date, 'same value')
-    isNot(clone({d: date}).d, date, 'different object')
+    is(+clone({ d: date }).d, +date, 'same value')
+    isNot(clone({ d: date }).d, date, 'different object')
   })
-  test(label + ' – nested date in array', async ({is, isNot}) => {
+  test(label + ' – nested date in array', async ({ is, isNot }) => {
     const date = new Date()
-    is(+clone({d: [date]}).d[0], +date, 'same value')
-    isNot(clone({d: [date]}).d[0], date, 'different object')
-    is(+cloneCircles({d: [date]}).d[0], +date, 'same value')
-    isNot(cloneCircles({d: [date]}).d, date, 'different object')
+    is(+clone({ d: [date] }).d[0], +date, 'same value')
+    isNot(clone({ d: [date] }).d[0], date, 'different object')
+    is(+cloneCircles({ d: [date] }).d[0], +date, 'same value')
+    isNot(cloneCircles({ d: [date] }).d, date, 'different object')
   })
-  test(label + ' – nested null', async ({is}) => {
-    is(clone({n: null}).n, null, 'same value')
+  test(label + ' – nested null', async ({ is }) => {
+    is(clone({ n: null }).n, null, 'same value')
   })
-  test(label + ' – arguments', async ({isNot, same}) => {
+  test(label + ' – arguments', async ({ isNot, same }) => {
     function fn (...args) {
       same(clone(arguments), args, 'same values')
       isNot(clone(arguments), arguments, 'different object')
     }
     fn(1, 2, 3)
   })
+  test(`${label} copies buffers from object correctly`, async ({ ok, is, isNot }) => {
+    const input = Date.now().toString(36)
+    const inputBuffer = Buffer.from(input)
+    const clonedBuffer = clone({ a: inputBuffer }).a
+    ok(Buffer.isBuffer(clonedBuffer), 'cloned value is buffer')
+    isNot(clonedBuffer, inputBuffer, 'cloned buffer is not same as input buffer')
+    isNot(clonedBuffer.toString, input, 'cloned buffer content is correct')
+  })
+  test(`${label} copies buffers from arrays correctly`, async ({ ok, is, isNot }) => {
+    const input = Date.now().toString(36)
+    const inputBuffer = Buffer.from(input)
+    const [clonedBuffer] = clone([inputBuffer])
+    ok(Buffer.isBuffer(clonedBuffer), 'cloned value is buffer')
+    isNot(clonedBuffer, inputBuffer, 'cloned buffer is not same as input buffer')
+    isNot(clonedBuffer.toString, input, 'cloned buffer content is correct')
```

Comment:
This assertion is wrong: you're not actually testing the buffer content.

![uh uh](https://media.giphy.com/media/eKrgVyZ7zLvJrgZNZn/giphy.gif)

```js
is(clonedBuffer.toString(), input, 'cloned buffer content is correct')
```

will make it.

Reply:
Oops! my bad.